src/domains/Game/Games.repository.ts

```diff
@@ -1,24 +1,50 @@
-import { Injectable, Logger } from '@nestjs/common';
+import { CACHE_MANAGER, Inject, Injectable, Logger } from '@nestjs/common';
+import { Cache } from 'cache-manager';
 import { Game } from './Game.aggregate';
-import { EventStore, StoreEventBus } from '@peterdijk/nestjs-eventstoredb';
+import { EventStore } from '@peterdijk/nestjs-eventstoredb';
+import { CACHE_KEYS } from '../../infrastructure/constants';
+import { instanceToPlain, plainToInstance } from 'class-transformer';
 
 @Injectable()
 export class GamesRepository {
   private readonly aggregate = 'game';
 
   constructor(
     private readonly eventStore: EventStore,
-    private readonly eventBus: StoreEventBus,
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
   ) {}
   private logger = new Logger(GamesRepository.name);
 
+  async updateOrCreate(game: Game): Promise<void> {
+    const cacheKey = this.getCacheKey(game.id);
+    const serializedGame = JSON.stringify(instanceToPlain(game));
+    // return this.cacheManager.set(cacheKey, serializedGame);
```

Comment:
while commented out, `findOneById` flow build up the aggregate from all past events. Slow but working for now. Storing Aggregate object in cache doesn't seem possible in Node ?